---
geometry:
- margin=0.75in
linestretch: 1.15
mainfont: Helvetica
fontsize: 11pt
linkcolor: blue
---

\setlist{leftmargin=15pt,listparindent=0pt,labelwidth=0pt,itemindent=0em,labelindent=0em}

\vskip-0.5cm
  
\hfill Sep 20, 2023 \vskip-0.5cm

Editorial Board  
*Nucleic Acids Research*

\vskip0.3cm

Dear editor,

We are pleased to submit our manuscript *‘Massively parallel identification of sequence motifs triggering ribosome-associated mRNA quality control’* for consideration at *Nucleic Acids Research*. 

**What was previously known.**
Recent work in the translation field has implicated a key role for the protein coding sequence in regulating stability of eukaryotic mRNAs.
Studies in this area to date have relied on the use of single reporter constructs or genome-wide studies on endogenous mRNAs.
However, both these approaches cannot identify the sequence features that trigger mRNA decay, genome-wide studies being correlational while single reporter constructs are limited in the number of sequences that can be tested.
Thus, the field is lacking a way to systematically and comprehensively identify sequences that trigger mRNA decay and the mechanisms by which they do so.

**Technological and scientific advances in this work.**
In this study, we develop and apply a massively parallel approach to experimentally test thousands of sequence motifs for their effect on mRNA stability in *S. cerevisiae*.
Using this approach, our work reports on the following key discoveries:
1. We identify a large set of dipeptide sequence motifs that trigger co-translational RNA decay.
2. We find that several of these motifs trigger mRNA decay through the ribosome-associated mRNA quality control (RQC) pathway.
3. We use a deep mutational scanning approach to identify the amino acid characteristics that confer mRNA instability and RQC dependence.
4. Measurements using our massively parallel assay accurately predict the effect of endogenous sequences on mRNA stability.

**Significance of our findings.**
Our results will be of immediate interest to the translation field as they provide a comprehensive set of sequence motifs that can be used to interpret ribosome profiling data and identify ribosome stalling sequences that trigger mRNA decay.
These motifs will also be of practical value to researchers in the RNA therapeutics area to design synthetic mRNA sequences that are stable.
Finally, the massively parallel reporter assay and the deep mutational scanning approach we present here will be of broad utility to RNA researchers to assess the impact of sequence variation on mRNA stability.

Thank you for considering our manuscript for publication in *Nucleic Acids Research*.

Sincerely,  
![](rasi_sign.png){width=60px}  
Arvind Rasi Subramaniam
